### PR TITLE
feat(quickjs): add swarm task tool

### PR DESCRIPTION
**Pull request by colifran**
### Summary

Adds the swarm task tool to @langchain/quickjs and fixes skill module loading for the agentskills.io spec, enabling parallel subagent dispatch from inside the QuickJS code interpreter.

Swarm task tool (tools/swarm-task.ts) is a PTC-only tool that bridges the code interpreter to the host runtime for subagent dispatch. It supports two modes:
1. Direct model calls with structured output (invoke mode) for classification/extraction tasks
2. Full agentic loops with tools (agent mode) for multi-step reasoning. It handles schema normalization, model variant caching with TTL-based sweeping, and structured output binding.

Spec-compliant entrypoint resolution — loadSkill() now resolves the entrypoint from metadata.entrypoint (the agentskills.io extension point) first, falling back to the legacy top-level module field. This lets skills use spec-compliant frontmatter without a top-level module key.

Subdirectory entrypoint support — the QuickJS module normalizer now uses the entrypoint's directory (not the skill root) when resolving relative imports from a bare specifier. This fixes import { ... } from "./sibling.js" when the entrypoint lives in a subdirectory like scripts/index.ts.

Supporting changes include wiring skills metadata through the middleware and session layers, extending the transform pipeline to strip import type/export type declarations, adding requiredPtcTools validation, and session improvements (extractToolText, stripLineNumbers, __sessionId__).

The swarm skill itself (table management, batching, filtering, interpolation) is distributed separately and not included here.

### Tests
- 41 tests for the swarm task tool covering both dispatch modes, schema normalization, variant caching and TTL sweeping, error handling, and edge cases around model binding
- 3 new tests for entrypoint resolution (metadata.entrypoint preferred over module, subdirectory relative imports)

### CODE REVIEW

File: libs/providers/quickjs/src/in-memory-backend.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+ * `<root>/foo/bar.ts` becomes `/foo/bar.ts`.
+ */
+function readDirectory(
+  rootDir: string,
+  currentDir: string,
+  files: Map<string, string>,
+): void {
+  for (const entry of fs.readdirSync(currentDir)) {
+    const fullPath = path.join(currentDir, entry);
+    const stat = fs.statSync(fullPath);
```
[the remark this answers is not in the document]
Comment (colifran) [reply libs/providers/quickjs/src/in-memory-backend.ts]: This is valid. I just pushed an update to fix this.

[the remark this answers is not in the document]
Comment (colifran) [reply libs/providers/quickjs/src/in-memory-backend.ts]: Is it resolved?

**Review by hntrl — APPROVED** [review]
